### ISSUE TITLE
Future of scan.nextcloud.com

### ISSUE THREAD

**Issue by tflidd** [nextcloud/nextcloud.com#574]
What is the status of the security scanner?

The current status is not satisfying, it just does not work. Is there a fix to be expected soon, if it takes longer, we should probably state that there is maintenance work going on. Or if we don't want to maintain it any more, I'd stop promoting this feature.

ref. https://help.nextcloud.com/t/nextcloud-security-scan-scan-failed/18579

@LukasReschke

**Comment by JMCA2** [comment]
Related issue: #535

**Comment by arucard21** [comment]
For what it's worth, I looked into this for a bit. It seems that the scanner works by sending a POST request to an API on the scan.nextcloud.com server. This API seems to return 201 if it's still ongoing and 200 if it's complete. The scanner checks for completion 5 additional times, all of which still seem to return 201 so it times out. Even if I resend the request from my browser's developer tools after waiting a bit, it keeps returning 201. 

The reason it looks like the scanner is working for cloud.nextcloud.org is that it skips the actual scan and just directly retrieves the results of a scan, presumably an older scan.

**Comment by mcteefax** [comment]
I've tried several instances like https://host.domian.de**/nextcloud** and it works.
When I use an URL without virtual directory like https.//host.domain.de it doesn't work.
Even Trying to use "Redirect "/nextcloud" "https://host.domain.de/" in the apache-config no effort.

I hope that the problem will be solved soon.

By the way: I really like the scanner and it's very useful anyway!!!!!! Thanks for it!!!!!

**Comment by arucard21** [comment]
I can confirm what @mcteefax says. I just tried it with my own Nextcloud instance, by adding a redirect from `/nextcloud` on my main domain to my cloud sub-domain. The scanner worked correctly with the redirected URL, but not with the URL for my cloud sub-domain. 

There seems to be something going wrong in the POST request to `/api/queue`. Is the code for this API available? I couldn't find it on Github.

**Comment by mcteefax** [comment]
Hi @arucard21 ,
how did you make the redirect?
My way was:
1. create a directory: /var/www/html/**next** and adapt the filepermissions
2. create .htaccess in the new directory and adapt the filepermissions
3. Edit .htaccess an paste 
`Redirect "/next" "https://host.domain.de/`

In my case, it doesn't work. Your config would be a **very helpful workaround**.

**Comment by arucard21** [comment]
@mcteefax I actually just created a symlink in my filesystem. I just wanted to quickly check if it made a difference. 

My setup is like this:
`/var/www/html/` -> this is accessible from my main domain, call it `https://my.domain.com`
`/var/www/nextcloud/` -> this is accessible from my cloud sub-domain, which would be `https://could.my.domain.com`

Both of these paths are configured as different virtualhosts in apache. So I just created a symlink in `/var/www/html`. 
```
cd /var/www/html/
ln -s ../nextcloud
```
I also tried creating this symlink in `/var/www/nextcloud/` itself, but I didn't get this working in the scanner. I'm not sure if this just didn't work, or if I had reached the scan limit, or if the scanner automatically looks for the `/nextcloud` path and ended up in an infinite loop. But you might give it a try as well. I'm not sure how stable or safe this symlink approach is though.
